CrossMod/GUI/BGSettings.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+namespace CrossMod.GUI
+{
+    public partial class BGSettings : Form
+    {
+
+        public static float red { get; set; } = 0.25f;
+        public static float green { get; set; } = 0.25f;
+        public static float blue { get; set; } = 0.25f;
+        public static bool renderAlpha { get; set; } = false;
+
+        public BGSettings()
+        {
+            InitializeComponent();
+        }
+
+        private void RedBox_TextChanged(object sender, EventArgs e)
+        {
+            if (System.Text.RegularExpressions.Regex.IsMatch(redBox.Text, "[^0-9]"))
+            {
+                redBox.Text = redBox.Text.Remove(redBox.Text.Length - 1);
+            }
+            if (redBox.Text == "")
+            {
+                redBox.Text = "0";
+            }
+            if (Int32.Parse(redBox.Text) >= 256)
+            {
+                redBox.Text = "255";
+            }
+            red = float.Parse(redBox.Text) / 255.0f;
+        }
+
+        private void GreenBox_TextChanged(object sender, EventArgs e)
+        {
+            if (System.Text.RegularExpressions.Regex.IsMatch(greenBox.Text, "[^0-9]"))
```

Comment:
This can be simplified to just `float.TryParse`. If the method returns false, the parsing failed. There's no need for the RGB values to be clamped to 1f.

Reply:
I'm still somewhat new to VS, I'll keep that in mind!